src/viam/sdk/resource/resource_manager.cpp

```diff
@@ -35,7 +35,8 @@ std::shared_ptr<Resource> ResourceManager::resource(std::string name) {
             return resources_.at(short_name);
         }
     }
-    throw std::runtime_error("Unable to find resource named " + name);
```

Comment:
This was a contract violation with the existing style of code which expects to check the return value of `ResourceManager::resource` rather than dealing with an exception. We can reverse that decision but it will require updating all the existing callers of `ResourceManager::resource`.

Reply:
[opt] I think I wrote the examples back when throwing was the default behavior for a missing resource. Might be a good idea to add nullptr checks for those (maybe in a different PR)